service/src/test/java/bio/terra/profile/pact/provider/BPMProviderTest.java

```diff
@@ -124,6 +142,111 @@ Map<String, Object> azureBillingProfileState() throws InterruptedException {
         "managedResourceGroupId", profile.managedResourceGroupId().get());
   }
 
+  @State("two billing profiles exist")
+  void twoProfilesExistState() throws InterruptedException {
+    var profilesIds =
+        List.of(
+            ProviderStateData.azureBillingProfile.id(), ProviderStateData.gcpBillingProfile.id());
+    when(samService.listProfileIds(any())).thenReturn(profilesIds);
+    when(profileDao.listBillingProfiles(anyInt(), anyInt(), eq(profilesIds)))
+        .thenReturn(
+            List.of(ProviderStateData.azureBillingProfile, ProviderStateData.gcpBillingProfile));
+  }
+
+  @State("a managed app exists")
+  Map<String, Object> managedAppExistsState() throws InterruptedException {
+    var profile = ProviderStateData.azureBillingProfile;
+    var subscriptionId = profile.subscriptionId().get();
+    when(azureService.getAuthorizedManagedAppDeployments(eq(subscriptionId), eq(false), any()))
+        .thenReturn(
+            List.of(
+                new AzureManagedAppModel()
+                    .subscriptionId(subscriptionId)
+                    .applicationDeploymentName("appDeploymentName")
+                    .region("dummyRegion")
+                    .assigned(false)
+                    .managedResourceGroupId(profile.managedResourceGroupId().get())
+                    .tenantId(profile.tenantId().get())));
+    return Map.of("subscriptionId", subscriptionId.toString());
+  }
+
+  @State("a JobService that supports profile creation")
+  Map<String, Object> creationJobServiceExistsState() {
+    var jobBuilder = mock(JobBuilder.class);
+
+    var profile = ProviderStateData.azureBillingProfile;
+
+    when(jobService.newJob()).thenReturn(jobBuilder);
```

Comment:
I copied this mocking from the unittest for profile creation. I first tried to create a more "real" JobService so that more of the actual flight code would execute, but that proved to be impossible with all the of Stairway dependencies.

Reply:
I went down the same path, with the same result